### PR TITLE
ncurses6: update to version 6.4.

### PR DESCRIPTION
**Pull request by OscarL**
Also, removed older recipe/patch.

### CODE REVIEW

**Comment by korli** [comment]
It seems 6.4 is still relevant for CVE-2023-29491 ...  
Arch uses https://invisible-mirror.net/archives/ncurses/current/ncurses-6.4-20230520.tgz along with --disable-root-access
    --disable-root-environ

**Comment by korli** [comment]
Should be now https://invisible-mirror.net/archives/ncurses/6.4/ncurses-6.4-20230520.tgz

**Comment by jmairboeck** [comment]
If you want to use something a bit more official: the author also provides the snapshots on GitHub, which as them all tagged: https://github.com/ThomasDickey/ncurses-snapshots/tags. The one in question is available here: https://github.com/ThomasDickey/ncurses-snapshots/archive/refs/tags/v6_4_20230520.tar.gz

**Comment by korli** [comment]
> @korli any reason for going for that particular date "20230520", over just using the more recent "20240330"?

Who else uses these newer snapshots?

**Comment by korli** [comment]
Then 20230520 should be OK?

**Comment by korli** [comment]
> Still not sure why the need to follow Arch here instead of other distros, but... fine... I'll use that.

If Arch didn't see the need to update, then it must work good enough. With rolling distros it can be adventurous to be in the firsts to updates. IMO being cutting-edge needs more workforce, which we don't have.

**Comment by korli** [comment]
Looks good. Just wondering about the version naming 6.4.20230520. Maybe 6.4_20230520 would be better.

File: sys-libs/ncurses/ncurses6-6.4_20230520.recipe [hunk 1]
```diff
@@ -3,28 +3,31 @@ DESCRIPTION="The Ncurses (new curses) library is a free software emulation of \
...
+	1998-2017,2018 Free Software Foundation, Inc."
 LICENSE="MIT"
-REVISION="2"
-SOURCE_URI="https://ftpmirror.gnu.org/ncurses/ncurses-$portVersion.tar.gz
-	https://ftp.gnu.org/gnu/ncurses/ncurses-$portVersion.tar.gz"
-CHECKSUM_SHA256="97fc51ac2b085d4cde31ef4d2c3122c21abc217e9090a43a30fc5ec21684e059"
-SOURCE_DIR="ncurses-$portVersion"
-PATCHES="ncurses-$portVersion.patchset"
+REVISION="1"
+srcVersion=${portVersion//./_}
```
Comment (korli) [inline sys-libs/ncurses/ncurses6-6.4_20230520.recipe:21]: is it still needed?

Comment (korli) [reply sys-libs/ncurses/ncurses6-6.4_20230520.recipe:21]: OK, I wasn't sure.

**Comment by korli** [comment]
LGTM

**Pull request merged**